src/index.js

```diff
@@ -188,7 +188,7 @@ export class EditorView {
   // :: () → bool
   // Query whether the view has focus.
   hasFocus() {
-    return this.root.activeElement == this.dom
+    return this.dom.contains(this.root.activeElement)
```

Comment:
This will also return true if the focus is in a completely unrelated descendant element (for example a textarea in a node view). In general, the conditions under which something should be considered part of the same editor are fuzzy, and I'd rather stick to a simple model than introduce confusion here.

Reply:
I understand your position, i'm just sharing ideas. For now i'm happy with my "crazy" patch: it keeps view focused even if going into an island inside view; and it's meant to be used with https://github.com/pageboard/pagecut/blob/1017f8c9263414de17428ea6c2614a1574b0fbbf/src/specs.js#L388-L396.